sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2263,8 +2263,6 @@ object SQLConf {
         "It was removed to prevent loosing of users data for non-default value."),
       RemovedConfig("spark.sql.legacy.compareDateTimestampInTimestamp", "3.0.0", "true",
         "It was removed to prevent errors like SPARK-23549 for non-default value."),
-      RemovedConfig("spark.sql.variable.substitute.depth", "3.0.0", "40",
```

Comment:
Could we add the description in line 2255 and explain which removed SQLConfigs should NOT be added to removedSQLConfigs?

Reply:
Here is the PR https://github.com/apache/spark/pull/27653